### PR TITLE
Update base64 dependency to 0.21.0

### PR DESCRIPTION
**Pull request by brendandburns**
- [X] This is related to #5678 there's a dependency mis-match that between some libraries I need for HTTP and the (old) base64 crate in wasmtime.
- [X] Revs the base64 dependency to 0.21.0, small API change.
- [X] This PR contains test cases, if meaningful. Not necessary, covered by existing tests.

### CODE REVIEW

**Comment by pchickey** [comment]
Agree we should update this dep, but I need to learn how to use `cargo vet` to generate the correct sort of record for the diff. I'll do my best to take care of that before eod. Our project policy is that only core contributors are trusted to vet packages, but I have never actually done it myself since that policy was set a few months ago.

**Comment by pchickey** [comment]
Please rebase on main after #5707 merges, and remove the last commit modifying the vet toml. Thanks!

**Comment by brendandburns** [comment]
@pchickey rebased, thanks for taking care of the vet part.

**Comment by brendandburns** [comment]
So, I dropped my cargo vet related commit, perhaps I wasn't supposed to do that? Anyway, I'm still seeing this cargo vet error:

```
Run cargo vet --locked
Vetting Failed!

1 unvetted dependencies:
  base64:0.21.0 missing ["safe-to-deploy"]
Error: Process completed with exit code 255.
```

Let me know the right way to fix this.

Thanks!

**Comment by pchickey** [comment]
Nope, thats me making a mistake on the vetting. I'll fix it

**Comment by pchickey** [comment]
https://github.com/bytecodealliance/wasmtime/pull/5710 will fix the vet problem, someone will review it monday. Sorry!

**Comment by pchickey** [comment]
I checked a rebase on main locally and this should be fixed now

**Review by peterhuene — APPROVED** [review]

**Comment by brendandburns** [comment]
Thanks @pchickey for the fixes and @peterhuene for the quick merge!

**Pull request merged**
